servers/src/main/java/tachyon/worker/block/BlockWorker.java

```diff
@@ -0,0 +1,281 @@
+    mWorkerMetricsSystem.start();
+
+    // Add the metrics servlet to the web server, this must be done after the metrics system starts
+    mWebServer.addHandler(mWorkerMetricsSystem.getServletHandler());
+
+    mSyncExecutorService.submit(mBlockMasterSync);
+    mWebServer.startWebServer();
+    mThriftServer.serve();
+  }
+
+  /**
+   * Stops the block worker. This method should only be called to terminate the worker.
+   *
+   * @throws IOException if the data server fails to close.
+   */
+  public void stop() throws IOException {
+    mDataServer.close();
+    mThriftServer.stop();
+    mThriftServerSocket.close();
+    mBlockMasterSync.stop();
+    mSyncExecutorService.shutdown();
+    try {
+      mWebServer.shutdownWebServer();
+    } catch (Exception e) {
+      LOG.error("Failed to stop web server", e);
+    }
+    mBlockDataManager.stop();
+    while (!mDataServer.isClosed() || mThriftServer.isServing()) {
+      // TODO: The reason to stop and close again is due to some issues in Thrift.
+      mThriftServer.stop();
+      mThriftServerSocket.close();
+      CommonUtils.sleepMs(null, 100);
+    }
+  }
+
+  /**
+   * Helper method to create a {@link org.apache.thrift.server.TThreadPoolServer} for handling
+   * incoming RPC requests.
+   *
+   * @return a thrift server
+   */
+  private TThreadPoolServer createThriftServer() {
+    int minWorkerThreads =
+        mTachyonConf.getInt(Constants.WORKER_MIN_WORKER_THREADS, Runtime.getRuntime()
+            .availableProcessors());
+    int maxWorkerThreads =
+        mTachyonConf.getInt(Constants.WORKER_MAX_WORKER_THREADS,
+            Constants.DEFAULT_WORKER_MAX_WORKER_THREADS);
+    WorkerService.Processor<BlockServiceHandler> processor =
+        new WorkerService.Processor<BlockServiceHandler>(mServiceHandler);
+    return new TThreadPoolServer(new TThreadPoolServer.Args(mThriftServerSocket)
+        .minWorkerThreads(minWorkerThreads).maxWorkerThreads(maxWorkerThreads).processor(processor)
+        .transportFactory(new TFramedTransport.Factory())
+        .protocolFactory(new TBinaryProtocol.Factory(true, true)));
+  }
+
+  /**
+   * Helper method to create a {@link org.apache.thrift.transport.TServerSocket} for the RPC server
+   *
+   * @return a thrift server socket
+   */
+  private TServerSocket createThriftServerSocket() {
+    try {
+      return new TServerSocket(getWorkerAddress());
+    } catch (TTransportException tte) {
+      LOG.error(tte.getMessage(), tte);
+      throw Throwables.propagate(tte);
+    }
+  }
+
+  /**
+   * Helper method to get the {@link java.net.InetSocketAddress} of the worker.
+   *
+   * @return the worker's address
+   */
+  private InetSocketAddress getWorkerAddress() {
+    String workerHostname = NetworkUtils.getLocalHostName(mTachyonConf);
+    int workerPort = mTachyonConf.getInt(Constants.WORKER_PORT, Constants.DEFAULT_WORKER_PORT);
+    return new InetSocketAddress(workerHostname, workerPort);
+  }
+
+  // For unit test purposes only
+
```

Comment:
Originally there was going to be more than one method, but now the space doesn't make sense. I've removed it.